app/managers/performance_metrics_manager.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getTimeSinceStartup} from 'react-native-startup-time';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'channelSwitch' |
+    'teamSwitch';
+
+class PerformanceMetricsManager {
+    private startTimes: {[metricName: string]: undefined | number} = {};
+    private target: Target;
+
+    public setLoadTarget(target: Target) {
+        this.target = target;
+    }
+
+    public finishLoad(location: Target) {
+        if (this.target !== location) {
+            return;
+        }
+
+        getTimeSinceStartup().then((v) => console.log(`Took ${v}`));
+    }
+
+    public startMetric(metricName: MetricName) {
+        this.startTimes[metricName] = Date.now();
+    }
+
+    public endMetric(metricName: MetricName) {
+        const startTime = this.startTimes[metricName];
+        if (startTime === undefined) {
+            return;
```

Comment:
Do we want to log anything if this happens? I want to do that in the web app, but I haven't yet because of how complicated navigation is there

Reply:
So... back to this... `performance.mark` is not available. Only `performance.now` :sweat_smile: 

So... right now the code you see is the "final" state. Let me know if you would change anything.